core/src/test/java/org/ehcache/core/config/BaseCacheConfigurationTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core.config;
+
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.ResourceType;
+import org.ehcache.config.SizedResourcePool;
+import org.ehcache.config.units.EntryUnit;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.ehcache.config.ResourceType.Core.HEAP;
+
+/**
+ * BaseCacheConfigurationTest
+ */
+public class BaseCacheConfigurationTest {
+
+  @Test(expected = NullPointerException.class)
+  public void testThrowsWithNullKeyType() {
+    new BaseCacheConfiguration<Object, String>(null, String.class, null,
+      null, null, getValidResourcePools());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testThrowsWithNullValueType() {
+    new BaseCacheConfiguration<Long, Object>(Long.class, null, null,
+      null, null, getValidResourcePools());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testThrowsWithNullResourcePools() {
+    new BaseCacheConfiguration<Long, String>(Long.class, String.class, null,
+      null, null, null);
+  }
+
+  private ResourcePoolsImpl getValidResourcePools() {
```

Comment:
Returning a ResourcePools would be enough. Also, you don't really need a fully initialized pool. `return mock(ResourcePools.class)` would work perfectly I think.